### PR TITLE
Stop trapping console.error & console.warn & expand Vue error handling.

### LINKED ISSUE #1344: console_error_throw.js obscures failing unit tests
would be nice to remove this so you can debug broken tests more easily.

### PR DESCRIPTION
**Pull request by NodeGuy**
Closes #1344

Description:

This is an experiment to see if it improves error handling and whether it is easier to diagnose from where errors are coming.

### CODE REVIEW

File: app/src/renderer/main.js [hunk 1]
```diff
@@ -30,12 +30,26 @@ window.addEventListener(`unhandledrejection`, function(event) {
 window.addEventListener(`error`, function(event) {
   Raven.captureException(event.reason)
 })
-Vue.config.errorHandler = error => {
+
+Vue.config.errorHandler = (error, vm, info) => {
+  console.error(`An error has occurred: ${error}
+
+Guru Meditation #${info}`)
```
Comment (jbibla) [inline app/src/renderer/main.js]: wut

Comment (NodeGuy) [reply app/src/renderer/main.js]: `Guru Meditation #` is what Amiga computers displayed when they crashed.  I'm just showing off how old I am.  😉

**Comment by NodeGuy** [comment]
How about throw only when `config.devMode` is true?

**Comment by faboweb** [comment]
Sounds good @NodeGuy

**Pull request merged**